### ISSUE TITLE
 polybar-msg cmd restart causes systray to disappear until second restart

### ISSUE THREAD

**Issue by TheBabu** [polybar/polybar#2035]
When every I use the command `polybar-msg cmd restart` it restarts the polybar as expected, but it the systray on the bottom right seems to disappear (without any of the processes going away). It only seems to reappear after the running the commmand a second time.

Normal systray:
![Screenshot_2020-03-08_11-01-13](https://user-images.githubusercontent.com/24832107/76169364-a2752000-6134-11ea-93b3-b26bc2d0b804.png)
Disappeared systray:
![Screenshot_2020-03-08_11-01-21](https://user-images.githubusercontent.com/24832107/76169356-97ba8b00-6134-11ea-879e-3e102134f7ae.png)

**Comment by patrick96** [comment]
What WM?

**Comment by TheBabu** [comment]
This issue is occurring on i3.

**Comment by patrick96** [comment]
Can you post the entire polybar log when this happens. My guess is that the tray is positioned behind the bar by i3. Please also post the bar section of your config.

**Comment by TheBabu** [comment]
This is the output of `polybar bottom &` after doing `polybar-msg cmd restart` a couple of times.
```warn: Termination signal received, shutting down...                                                                                                                         1   879  12:12:58
warn: module/i3: Attempting to reconnect socket (reason: Unexpected EOF while reading header)
warn: Termination signal received, shutting down...                                                                                                                         1   879  12:13:01
warn: module/i3: Attempting to reconnect socket (reason: Unexpected EOF while reading header)
error: tray: Failed to put tray above 0x0c00002 in the stack (XCB_MATCH (8))                                                                                                1   879  12:13:05
warn: Termination signal received, shutting down...                                                                                                                         1   879  12:13:05
warn: module/i3: Attempting to reconnect socket (reason: Unexpected EOF while reading header)
error: tray: Failed to put tray above 0x0c00002 in the stack (XCB_MATCH (8))                                                                                                1   879  12:13:07
warn: Termination signal received, shutting down...                                                                                                                         1   879  12:13:07
warn: module/i3: Attempting to reconnect socket (reason: Unexpected EOF while reading header)
error: tray: Failed to put tray above 0x0c00002 in the stack (XCB_MATCH (8))                                                                                                1   879  12:13:07
warn: Termination signal received, shutting down...                                                                                                                         1   879  12:13:10
warn: module/i3: Attempting to reconnect socket (reason: Unexpected EOF while reading header)
error: tray: Failed to put tray above 0x0c00002 in the stack (XCB_MATCH (8))                                                                                                1   879  12:13:13
warn: Termination signal received, shutting down...                                                                                                                         1   879  12:13:13
warn: module/i3: Attempting to reconnect socket (reason: Unexpected EOF while reading header)
error: tray: Failed to put tray above 0x0c00002 in the stack (XCB_MATCH (8)) ```
This is my config for my bottom bar (where my systray is located):
```;Bottom Bar
[bar/bottom]
monitor = ${env:MONITOR:HDMI1}
monitor-fallback = eDP1
bottom = true
width = 100%
height = 35

background = ${bar/top.background}
foreground = ${bar/top.foreground}

line-color = ${bar/top.background}
line-size = 2

spacing = 3
padding-right = 4
module-margin-left = 0
module-margin-right = 6

font-0 = NotoSans-Regular:size=10;2
font-1 = unifont:size=8;-1
font-2 = FontAwesome:size=10;3
font-3 = NotoSans-Regular:size=10;1
font-4 = MaterialIcons:size=12;1
font-5 = Termsynu:size=10:antialias=false;2

modules-left = i3
modules-right = cpu memory
tray-position = right
tray-maxsize = 24

enable-ipc = true
```

**Comment by patrick96** [comment]
I think you are suffering from the same thing as reported in #425, just the other way around.

I think the solution here might be to set `override-redirect = true`. This will cause the bar to overlap with the other windows though, there is a section on our [known issues page](https://github.com/polybar/polybar/wiki/Known-Issues#override-redirect-in-i3) addressing this.

**Comment by patrick96** [comment]
This should be fixed by #2609 if you are using the new tray module.
